### PR TITLE
Enable pass version of lint tool via environment varaible & upgrade oav versoin

### PR DESCRIPTION
**Pull request by jianyexi**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -394,4 +394,26 @@ export const getOpenapiType = async function(configFile: string):Promise<string>
     return types.indexOf(type) !== -1 ;
   }
   
+}
+
+interface LintVersion {
```
Comment (ruowan) [inline src/utils.ts]: prefer to use type.

File: src/momentOfTruth.ts [hunk 2]
```diff
@@ -21,8 +21,16 @@ async function getLinterResult(swaggerPath: string|null|undefined) {
     }
 
     let openapiType = await utils.getOpenapiType(swaggerPath);
+    let lintVersion = utils.getLinterVersion()
+    let lintVersionCmd = ''
+    if (lintVersion.classic) {
+        lintVersionCmd += ' --use=@microsoft.azure/classic-openapi-validator@' + lintVersion.classic + ' '
```
Comment (ruowan) [inline src/momentOfTruth.ts]: Can you move this space to cmd string join?

File: src/tests/momentOfTruthTest.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    
+    /**
+     * test if the lint tool run normally
+     */
+    @test @timeout(60000) async "TestGetLinterResult" () {
+      
+      process.env.CLASSIC_LINT_VERSION = "1.1.0"
+      process.env.LINT_VERSION = "1.1.0"
+      let result = await getLinterResult("./src/tests/Resource/swagger/test-lint-result.md");
+      assert.equal(Object.keys(result).length,3)
```
Comment (ruowan) [inline src/tests/momentOfTruthTest.ts]: assert content of result not only the length

File: src/utils.ts [hunk 4]
```diff
@@ -357,12 +357,12 @@ export const getOpenapiType = async function(configFile: string):Promise<string>
     console.log("parse failed with msg:" + err);
   }
 
-  if ( configFile.match(/.+\/specification\/.*\/resource-manager\/.*readme.md$/g)) {
+  if ( configFile.match(/.*specification\/.*\/resource-manager\/.*readme.md$/g)) {
```
Comment (ruowan) [inline src/utils.ts]: Why not just string.include('resource-manager')?

File: src/utils.ts [hunk 5]
```diff
@@ -394,4 +394,26 @@ export const getOpenapiType = async function(configFile: string):Promise<string>
...
   
+}
+
+interface LintVersion {
+  classic:string
+  present:string
+}
+
+export const getLinterVersion = function ():LintVersion {
+  let classicLintVersion =  process.env['CLASSIC_LINT_VERSION']
```
Comment (ruowan) [inline src/utils.ts]: If the vairbale not changed, use const if better. BTW, you can use arrow function instead assign function to a variable.

Comment (jianyexi) [reply src/utils.ts]: here , the variables will be changed

File: src/tests/momentOfTruthTest.ts (hunk 3 above)
Comment (jianyexi) [reply src/tests/momentOfTruthTest.ts]: done

File: src/utils.ts (hunk 1 above)
Comment (jianyexi) [reply src/utils.ts]: done

**Pull request merged**